### ISSUE TITLE
fs in env mongo

### ISSUE THREAD

**Issue by jeremykentbgross** [eslint/eslint#17588]
### Environment

Node version: v18.16.1
npm version: 9.5.1
Local ESLint version: 8.48.0
Global ESLint version: NA
Operating System: Ubuntu
mongod version: 6.0.8
mongosh version: 1.10.1

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
using the default
```
</details>

```js
/* eslint-env mongo */
fs.readFileSync(inPath, 'utf8');
```

### What did you expect to happen?

/* eslint-env mongo */ correctly allows the global function load(), but does not recognize that fs should also be available as expected and claims it is undefined.

See mongosh documentation: https://www.mongodb.com/docs/manual/reference/method/js-native/
(Or manually try it, it is there.)

### What actually happened?

The fs object definitely exists in the mongosh shell, but the /* eslint-env mongo */ directive does not seem to include it or recognize it, generating an no-undef error.

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50LWVudiBtb25nbyAqL1xuXG4vL21vbmdvIGdsb2JhbCBsb2FkIGlzIGF2YWlsYWJsZSBhbmQgY29ycmVjdGx5IGlkZW50aWZpZWRcbmxvYWQoXCJzb21lRmlsZS50eHRcIik7XG5cbi8vbW9uZ28gZ2xvYmFsIGZzIGlzIGF2YWlsYWJsZSBhbmQgTk9UIGNvcnJlY3RseSBpZGVudGlmaWVkXG5mcy5yZWFkRmlsZVN5bmMoXCJzb21lT3RoZXJGaWxlLnR4dFwiLCAndXRmOCcpO1xuXG4vL05vdGUgdGhhdCBmcyBpcyBhIGdsb2JhbGx5IGF2YWlsYWJsZSBvYmplY3QsIHRoYXQgaXMgYXZhaWxhYmxlIGluIG15IG1vbmdvc2ggc2hlbGw6XG4vL2h0dHBzOi8vd3d3Lm1vbmdvZGIuY29tL2RvY3MvbWFudWFsL3JlZmVyZW5jZS9tZXRob2QvanMtbmF0aXZlLyIsIm9wdGlvbnMiOnsiZW52Ijp7ImVzNiI6dHJ1ZX0sInBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fSwiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoic2NyaXB0In0sInJ1bGVzIjp7ImNvbnN0cnVjdG9yLXN1cGVyIjpbImVycm9yIl0sImZvci1kaXJlY3Rpb24iOlsiZXJyb3IiXSwiZ2V0dGVyLXJldHVybiI6WyJlcnJvciJdLCJuby1hc3luYy1wcm9taXNlLWV4ZWN1dG9yIjpbImVycm9yIl0sIm5vLWNhc2UtZGVjbGFyYXRpb25zIjpbImVycm9yIl0sIm5vLWNsYXNzLWFzc2lnbiI6WyJlcnJvciJdLCJuby1jb21wYXJlLW5lZy16ZXJvIjpbImVycm9yIl0sIm5vLWNvbmQtYXNzaWduIjpbImVycm9yIl0sIm5vLWNvbnN0LWFzc2lnbiI6WyJlcnJvciJdLCJuby1jb25zdGFudC1jb25kaXRpb24iOlsiZXJyb3IiXSwibm8tY29udHJvbC1yZWdleCI6WyJlcnJvciJdLCJuby1kZWJ1Z2dlciI6WyJlcnJvciJdLCJuby1kZWxldGUtdmFyIjpbImVycm9yIl0sIm5vLWR1cGUtYXJncyI6WyJlcnJvciJdLCJuby1kdXBlLWNsYXNzLW1lbWJlcnMiOlsiZXJyb3IiXSwibm8tZHVwZS1lbHNlLWlmIjpbImVycm9yIl0sIm5vLWR1cGUta2V5cyI6WyJlcnJvciJdLCJuby1kdXBsaWNhdGUtY2FzZSI6WyJlcnJvciJdLCJuby1lbXB0eSI6WyJlcnJvciJdLCJuby1lbXB0eS1jaGFyYWN0ZXItY2xhc3MiOlsiZXJyb3IiXSwibm8tZW1wdHktcGF0dGVybiI6WyJlcnJvciJdLCJuby1leC1hc3NpZ24iOlsiZXJyb3IiXSwibm8tZXh0cmEtYm9vbGVhbi1jYXN0IjpbImVycm9yIl0sIm5vLWV4dHJhLXNlbWkiOlsiZXJyb3IiXSwibm8tZmFsbHRocm91Z2giOlsiZXJyb3IiXSwibm8tZnVuYy1hc3NpZ24iOlsiZXJyb3IiXSwibm8tZ2xvYmFsLWFzc2lnbiI6WyJlcnJvciJdLCJuby1pbXBvcnQtYXNzaWduIjpbImVycm9yIl0sIm5vLWlubmVyLWRlY2xhcmF0aW9ucyI6WyJlcnJvciJdLCJuby1pbnZhbGlkLXJlZ2V4cCI6WyJlcnJvciJdLCJuby1pcnJlZ3VsYXItd2hpdGVzcGFjZSI6WyJlcnJvciJdLCJuby1sb3NzLW9mLXByZWNpc2lvbiI6WyJlcnJvciJdLCJuby1taXNsZWFkaW5nLWNoYXJhY3Rlci1jbGFzcyI6WyJlcnJvciJdLCJuby1taXhlZC1zcGFjZXMtYW5kLXRhYnMiOlsiZXJyb3IiXSwibm8tbmV3LXN5bWJvbCI6WyJlcnJvciJdLCJuby1ub25vY3RhbC1kZWNpbWFsLWVzY2FwZSI6WyJlcnJvciJdLCJuby1vYmotY2FsbHMiOlsiZXJyb3IiXSwibm8tb2N0YWwiOlsiZXJyb3IiXSwibm8tcHJvdG90eXBlLWJ1aWx0aW5zIjpbImVycm9yIl0sIm5vLXJlZGVjbGFyZSI6WyJlcnJvciJdLCJuby1yZWdleC1zcGFjZXMiOlsiZXJyb3IiXSwibm8tc2VsZi1hc3NpZ24iOlsiZXJyb3IiXSwibm8tc2V0dGVyLXJldHVybiI6WyJlcnJvciJdLCJuby1zaGFkb3ctcmVzdHJpY3RlZC1uYW1lcyI6WyJlcnJvciJdLCJuby1zcGFyc2UtYXJyYXlzIjpbImVycm9yIl0sIm5vLXRoaXMtYmVmb3JlLXN1cGVyIjpbImVycm9yIl0sIm5vLXVuZGVmIjpbImVycm9yIl0sIm5vLXVuZXhwZWN0ZWQtbXVsdGlsaW5lIjpbImVycm9yIl0sIm5vLXVucmVhY2hhYmxlIjpbImVycm9yIl0sIm5vLXVuc2FmZS1maW5hbGx5IjpbImVycm9yIl0sIm5vLXVuc2FmZS1uZWdhdGlvbiI6WyJlcnJvciJdLCJuby11bnNhZmUtb3B0aW9uYWwtY2hhaW5pbmciOlsiZXJyb3IiXSwibm8tdW51c2VkLWxhYmVscyI6WyJlcnJvciJdLCJuby11bnVzZWQtdmFycyI6WyJlcnJvciJdLCJuby11c2VsZXNzLWJhY2tyZWZlcmVuY2UiOlsiZXJyb3IiXSwibm8tdXNlbGVzcy1jYXRjaCI6WyJlcnJvciJdLCJuby11c2VsZXNzLWVzY2FwZSI6WyJlcnJvciJdLCJuby13aXRoIjpbImVycm9yIl0sInJlcXVpcmUteWllbGQiOlsiZXJyb3IiXSwidXNlLWlzbmFuIjpbImVycm9yIl0sInZhbGlkLXR5cGVvZiI6WyJlcnJvciJdfX19

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

I am able to get past it using the following:

/* eslint-disable-next-line no-undef */

I am therefore not blocked by it, however ideally eslint should know that fs is available due to the env

**Comment by Rec0iL99** [comment]
Hi @jeremykentbgross, thanks for the issue. `fs` is coming from `node` and is not a global variable. You need to `require` it.

```js
const fs = require("fs");
```

[playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50LWVudiBtb25nbywgbm9kZSAqL1xuXG5jb25zdCBmcyA9IHJlcXVpcmUoJ2ZzJylcblxuLy9tb25nbyBnbG9iYWwgbG9hZCBpcyBhdmFpbGFibGUgYW5kIGNvcnJlY3RseSBpZGVudGlmaWVkXG5sb2FkKFwic29tZUZpbGUudHh0XCIpO1xuXG4vL21vbmdvIGdsb2JhbCBmcyBpcyBhdmFpbGFibGUgYW5kIE5PVCBjb3JyZWN0bHkgaWRlbnRpZmllZFxuZnMucmVhZEZpbGVTeW5jKFwic29tZU90aGVyRmlsZS50eHRcIiwgJ3V0ZjgnKTtcblxuLy9Ob3RlIHRoYXQgZnMgaXMgYSBnbG9iYWxseSBhdmFpbGFibGUgb2JqZWN0LCB0aGF0IGlzIGF2YWlsYWJsZSBpbiBteSBtb25nb3NoIHNoZWxsOlxuLy9odHRwczovL3d3dy5tb25nb2RiLmNvbS9kb2NzL21hbnVhbC9yZWZlcmVuY2UvbWV0aG9kL2pzLW5hdGl2ZS8iLCJvcHRpb25zIjp7ImVudiI6eyJlczYiOnRydWV9LCJydWxlcyI6eyJjb25zdHJ1Y3Rvci1zdXBlciI6WyJlcnJvciJdLCJmb3ItZGlyZWN0aW9uIjpbImVycm9yIl0sImdldHRlci1yZXR1cm4iOlsiZXJyb3IiXSwibm8tYXN5bmMtcHJvbWlzZS1leGVjdXRvciI6WyJlcnJvciJdLCJuby1jYXNlLWRlY2xhcmF0aW9ucyI6WyJlcnJvciJdLCJuby1jbGFzcy1hc3NpZ24iOlsiZXJyb3IiXSwibm8tY29tcGFyZS1uZWctemVybyI6WyJlcnJvciJdLCJuby1jb25kLWFzc2lnbiI6WyJlcnJvciJdLCJuby1jb25zdC1hc3NpZ24iOlsiZXJyb3IiXSwibm8tY29uc3RhbnQtY29uZGl0aW9uIjpbImVycm9yIl0sIm5vLWNvbnRyb2wtcmVnZXgiOlsiZXJyb3IiXSwibm8tZGVidWdnZXIiOlsiZXJyb3IiXSwibm8tZGVsZXRlLXZhciI6WyJlcnJvciJdLCJuby1kdXBlLWFyZ3MiOlsiZXJyb3IiXSwibm8tZHVwZS1jbGFzcy1tZW1iZXJzIjpbImVycm9yIl0sIm5vLWR1cGUtZWxzZS1pZiI6WyJlcnJvciJdLCJuby1kdXBlLWtleXMiOlsiZXJyb3IiXSwibm8tZHVwbGljYXRlLWNhc2UiOlsiZXJyb3IiXSwibm8tZW1wdHkiOlsiZXJyb3IiXSwibm8tZW1wdHktY2hhcmFjdGVyLWNsYXNzIjpbImVycm9yIl0sIm5vLWVtcHR5LXBhdHRlcm4iOlsiZXJyb3IiXSwibm8tZXgtYXNzaWduIjpbImVycm9yIl0sIm5vLWV4dHJhLWJvb2xlYW4tY2FzdCI6WyJlcnJvciJdLCJuby1leHRyYS1zZW1pIjpbImVycm9yIl0sIm5vLWZhbGx0aHJvdWdoIjpbImVycm9yIl0sIm5vLWZ1bmMtYXNzaWduIjpbImVycm9yIl0sIm5vLWdsb2JhbC1hc3NpZ24iOlsiZXJyb3IiXSwibm8taW1wb3J0LWFzc2lnbiI6WyJlcnJvciJdLCJuby1pbm5lci1kZWNsYXJhdGlvbnMiOlsiZXJyb3IiXSwibm8taW52YWxpZC1yZWdleHAiOlsiZXJyb3IiXSwibm8taXJyZWd1bGFyLXdoaXRlc3BhY2UiOlsiZXJyb3IiXSwibm8tbG9zcy1vZi1wcmVjaXNpb24iOlsiZXJyb3IiXSwibm8tbWlzbGVhZGluZy1jaGFyYWN0ZXItY2xhc3MiOlsiZXJyb3IiXSwibm8tbWl4ZWQtc3BhY2VzLWFuZC10YWJzIjpbImVycm9yIl0sIm5vLW5ldy1zeW1ib2wiOlsiZXJyb3IiXSwibm8tbm9ub2N0YWwtZGVjaW1hbC1lc2NhcGUiOlsiZXJyb3IiXSwibm8tb2JqLWNhbGxzIjpbImVycm9yIl0sIm5vLW9jdGFsIjpbImVycm9yIl0sIm5vLXByb3RvdHlwZS1idWlsdGlucyI6WyJlcnJvciJdLCJuby1yZWRlY2xhcmUiOlsiZXJyb3IiXSwibm8tcmVnZXgtc3BhY2VzIjpbImVycm9yIl0sIm5vLXNlbGYtYXNzaWduIjpbImVycm9yIl0sIm5vLXNldHRlci1yZXR1cm4iOlsiZXJyb3IiXSwibm8tc2hhZG93LXJlc3RyaWN0ZWQtbmFtZXMiOlsiZXJyb3IiXSwibm8tc3BhcnNlLWFycmF5cyI6WyJlcnJvciJdLCJuby10aGlzLWJlZm9yZS1zdXBlciI6WyJlcnJvciJdLCJuby11bmRlZiI6WyJlcnJvciJdLCJuby11bmV4cGVjdGVkLW11bHRpbGluZSI6WyJlcnJvciJdLCJuby11bnJlYWNoYWJsZSI6WyJlcnJvciJdLCJuby11bnNhZmUtZmluYWxseSI6WyJlcnJvciJdLCJuby11bnNhZmUtbmVnYXRpb24iOlsiZXJyb3IiXSwibm8tdW5zYWZlLW9wdGlvbmFsLWNoYWluaW5nIjpbImVycm9yIl0sIm5vLXVudXNlZC1sYWJlbHMiOlsiZXJyb3IiXSwibm8tdW51c2VkLXZhcnMiOlsiZXJyb3IiXSwibm8tdXNlbGVzcy1iYWNrcmVmZXJlbmNlIjpbImVycm9yIl0sIm5vLXVzZWxlc3MtY2F0Y2giOlsiZXJyb3IiXSwibm8tdXNlbGVzcy1lc2NhcGUiOlsiZXJyb3IiXSwibm8td2l0aCI6WyJlcnJvciJdLCJyZXF1aXJlLXlpZWxkIjpbImVycm9yIl0sInVzZS1pc25hbiI6WyJlcnJvciJdLCJ2YWxpZC10eXBlb2YiOlsiZXJyb3IiXX0sInBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fSwiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoic2NyaXB0In19fQ==)

**Comment by jeremykentbgross** [comment]
@Rec0iL99 

Forgive me, perhaps my initial report was insufficiently clear, as it appears you have falsely assumed I was describing a completely different scenario/issue.

This is NOT running in the Node environment.  It is the MongoDB shell, and fs *IS* available there without any calls to require(). In fact no calls to require() of any kind are needed as my code works perfectly fine when imported using load() (as is shown in the provided sample) or simply typed into the shell, and fs apparently exists in the MongoDB Shell by default from the time it starts up without me importing it by use of either require() or load().

It sounds like you think I was having an issue with the CODE not working due to a lack of the require, but requiring fs or not isn't related to the issue I was reporting.

The REAL issue was that while fs is automatically as global variable that is available to me in my MongoDB shell from the moment it starts, just like load() is, AND my code actually works perfectly fine, ESLint doesn't recognize that it should.  The MongoDB environment declared to ESLint causes it to CORRECTLY recognize load() as a globally available function (which it is), but it also INCORRECTLY excludes fs, even though fs IS indeed available in the MongoDB Shell environment, and WITHOUT me making any calls to require() or load().

Hopefully that is clearer, but if not, it's not a big deal to me.  It's your project, I just figured that you would want it to accurately report for the MongoDB Shell, which it is not doing unless there is something unusual with my shell.

**Comment by mdjermanovic** [comment]
> /* eslint-env mongo */ correctly allows the global function load(), but does not recognize that fs should also be available as expected and claims it is undefined.
> 
> See mongosh documentation: https://www.mongodb.com/docs/manual/reference/method/js-native/ (Or manually try it, it is there.)

It looks to me that `mongo` and `mongosh` should be two different environments as these seem like two different shells with different globals, so I'm not sure if updating `mongo` with the new `mongosh` globals would be correct.

Either way, ESLint imports the list of `mongo` globals from the [`globals` package](https://www.npmjs.com/package/globals). If you think that the list is incorrect or incomplete, please file an issue in the `globals` repository:

https://github.com/sindresorhus/globals

**Comment by jeremykentbgross** [comment]
@mdjermanovic 

Thank you for the reply.

Regular mongo appears to be deprecated, leaving mongosh is the preferred way to interact with MongoDB outside of language APIs such as are on npm.  Regular mongo did not come with my download of MongoDB at all from the download center, and mongosh is the only one the download center provided independently, making me wonder if the legacy mongo is even still available at all.

So far I have not found any official source saying it is deprecated, but a web search produces multiple 3rd party outlets and bloggers saying that mongosh has superseded the older mongo, and even have screenshots of mongo itself reporting itself as superseded by mongosh with a warning at it's own startup on the CLI.

Under the circumstances, it would seem that the mongo environment for ESLint should actually refer to mongosh, or have a mongosh environment in addition to or as a replacement for the old mongo one.

Also for reference the globals npm package that you linked says that it no longer accepts additional environments, and that if one is needed for ESLint, they suggest to create a pluggin.  I am not familiar with this npm package, nor am I super motivated because of having only one suppressible error in my own more immediately pressing work.  I merely thought I would be dropping a helpful report to the maintainers of this project that ESLint isn't supporting the current standard.

For reference in case I am doubted, this is the output of auto complete on a brand new terminal with a brand new launch of mongosh:


```
~$ mongosh
Current Mongosh Log ID:	6515aa2bf185b5db48b50f7d
Connecting to:		mongodb://127.0.0.1:27017/?directConnection=true&serverSelectionTimeoutMS=2000&appName=mongosh+1.10.1
Using MongoDB:		6.0.8
Using Mongosh:		1.10.1

For mongosh info see: https://docs.mongodb.com/mongodb-shell/

------
   The server generated these startup warnings when booting
   2023-09-28T09:30:25.824-07:00: Using the XFS filesystem is strongly recommended with the WiredTiger storage engine. See http://dochub.mongodb.org/core/prodnotes-filesystem
   2023-09-28T09:30:27.612-07:00: Access control is not enabled for the database. Read and write access to data and configuration is unrestricted
   2023-09-28T09:30:27.612-07:00: This server is bound to localhost. Remote systems will be unable to connect to this server. Start the server with --bind_ip <address> to specify which IP addresses it should serve responses from, or with --bind_ip_all to bind to all interfaces. If this behavior is desired, start the server with --bind_ip 127.0.0.1 to disable this warning
   2023-09-28T09:30:27.612-07:00: Soft rlimits for open file descriptors too low
------

test> fs.	
fs.__proto__             fs.constructor           fs.hasOwnProperty        fs.isPrototypeOf         fs.propertyIsEnumerable  fs.toLocaleString
fs.toString              fs.valueOf

fs.Dir                   fs.Dirent                fs.F_OK                  fs.FileReadStream        fs.FileWriteStream       fs.R_OK
fs.ReadStream            fs.Stats                 fs.W_OK                  fs.WriteStream           fs.X_OK                  fs._toUnixTimestamp
fs.access                fs.accessSync            fs.appendFile            fs.appendFileSync        fs.chmod                 fs.chmodSync
fs.chown                 fs.chownSync             fs.close                 fs.closeSync             fs.constants             fs.copyFile
fs.copyFileSync          fs.cp                    fs.cpSync                fs.createReadStream      fs.createWriteStream     fs.exists
fs.existsSync            fs.fchmod                fs.fchmodSync            fs.fchown                fs.fchownSync            fs.fdatasync
fs.fdatasyncSync         fs.fstat                 fs.fstatSync             fs.fsync                 fs.fsyncSync             fs.ftruncate
fs.ftruncateSync         fs.futimes               fs.futimesSync           fs.lchmod                fs.lchmodSync            fs.lchown
fs.lchownSync            fs.link                  fs.linkSync              fs.lstat                 fs.lstatSync             fs.lutimes
fs.lutimesSync           fs.mkdir                 fs.mkdirSync             fs.mkdtemp               fs.mkdtempSync           fs.open
fs.openSync              fs.opendir               fs.opendirSync           fs.promises              fs.read                  fs.readFile
fs.readFileSync          fs.readSync              fs.readdir               fs.readdirSync           fs.readlink              fs.readlinkSync
fs.readv                 fs.readvSync             fs.realpath              fs.realpathSync          fs.rename                fs.renameSync
fs.rm                    fs.rmSync                fs.rmdir                 fs.rmdirSync             fs.stat                  fs.statSync
fs.symlink               fs.symlinkSync           fs.truncate              fs.truncateSync          fs.unlink                fs.unlinkSync
fs.unwatchFile           fs.utimes                fs.utimesSync            fs.watch                 fs.watchFile             fs.write
fs.writeFile             fs.writeFileSync         fs.writeSync             fs.writev                fs.writevSync

test> fs.
```

As you can see, fs is available by default.  If nobody cares, that's fine.  I was just trying to be helpful.  My apologies if I am wasting your time (and mine), as well as for not having the bandwidth at this time to explore making plugins or whatever has to happen to support what appears to be the new standard environment for MongoDB in ESLint.  I appreciate the existence of the ESLint package in any case.

-J
